cds/webapi/C#/BlazorCDS/BlazorCDS/Pages/EnvironmentChooser.razor

```diff
@@ -0,0 +1,178 @@
+                }
+            }
+        }
+    }
+
+    [Parameter]
+    public EventCallback<string> SelectedEnvUrlChanged { get; set; }
+
+    private InstanceCollection instances;
+
+    protected override async Task OnInitializedAsync()
+    {
+        try
+        {
+
+            var tokenResult = await TokenProvider.RequestAccessToken();
+
+            logger.LogInformation($"AccessTokenResult.Status: {tokenResult.Status}"); //RequiresRedirect
+            logger.LogInformation($"AccessTokenResult.RedirectUrl: {tokenResult.RedirectUrl}");
+
+            if (tokenResult.TryGetToken(out var token))
+            {
+                logger.LogInformation(token.ToString());
+            }
+            else
+            {
+                logger.LogInformation("No accesstoken");
+            }
+
+            //Creates an HttpClient based on the named definition found in Program.Main
+            var client = ClientFactory.CreateClient("GDSClient");
+
+            //Prepare the request to get the data
+            var request = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Get,
+                RequestUri = new Uri($"{client.BaseAddress}api/discovery/v2.0/Instances")
+            };
+
+            //Add the access token
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token.Value);
+            //Specify a JSON result is expected
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            //Send the request
+            var response = await client.SendAsync(request);
+
+            instances = new InstanceCollection();
+
+            if (response.IsSuccessStatusCode)
+            {
+                //Get the response content and parse it.
+                string result = response.Content.ReadAsStringAsync().Result;
+                JObject body = JObject.Parse(result);
+                JArray values = (JArray)body.GetValue("value");
+
+                if (!values.HasValues)
+                {
+                    var mockInstances = new Instance[] {
+                        new Instance() { FriendlyName = "Environment A", ApiUrl = "A" },
+                        new Instance() { FriendlyName = "Environment B", ApiUrl = "B" },
+                        new Instance() { FriendlyName = "Environment C", ApiUrl = "C" },
+                        };
+
+                    instances = new InstanceCollection() { value = mockInstances };
+                }
+
+                instances = new InstanceCollection() { value = JsonConvert.DeserializeObject<List<Instance>>(values.ToString()).ToArray() };
+
+                if (instances.value.Length > 0)
+                {
+                    AppState.SetEnvUrl(instances.value.First().ApiUrl);
+                }
+            }
+            else
+            {
+                //Something went wrong.
+            }
+        }
+        catch (Exception ex)
+        {
```

Comment:
The error should be handled here

Reply:
setting failedToLoad and showing a label instead of dropdown